### ISSUE TITLE
Implement color cycle

### ISSUE THREAD

**Issue by Datseris** [JuliaPlots/Makie.jl#189]
One of the amazing things of PyPlot is the existence of a default "color cycle".

In makie:
```
plot(rand(10), rand(10))
plot!(rand(10), rand(10))
plot!(rand(10), rand(10))
plot!(rand(10), rand(10))
```
produces 4 lines with black color, while in PyPlot each line that is not given a color explicitly, gets one from the default color cycle and advances the cycler timer by one.

This timer must be an attribute of the current axis one plots on.

**Comment by mkborregaard** [comment]
👍 GR, PlotlyJS and Plots also do that.

**Comment by SimonDanisch** [comment]
Yeah, there is a pretty simple way to implement that, which I disliked and got rid of :D 
I planned to recover it as part of the series recipe!

**Comment by mschauer** [comment]
Putting this into a recipe is great!

**Comment by mkborregaard** [comment]
https://github.com/JuliaPlots/AbstractPlotting.jl/pull/37

**Comment by lhupe** [comment]
Any updates on this issue? This is a major reason for me to stay with Plots for now — if I plot tens of data series, I don't want to have to assign colors by hand. 
As far as I can tell, JuliaPlots/AbstractPlotting.jl#37 doesn't work anymore (at least for me, copy-and-pasting the example to a clean REPL produced a gray blob) and there is no mention of color cycles in the docs.
